plugins/hydra_ray_launcher/hydra_plugins/ray_launcher/ray_launcher.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+import functools
+
+from omegaconf import open_dict
+from omegaconf import OmegaConf
+from omegaconf import DictConfig
+
+from hydra._internal.config_search_path import ConfigSearchPath
+from hydra._internal.pathlib import Path
+from hydra.plugins import Launcher
+from hydra.plugins import SearchPathPlugin
+from hydra.plugins.common.utils import (
+    configure_log,
+    filter_overrides,
+    run_job,
+    setup_globals,
+    HydraConfig,
+)
+
+import ray
+
+log = logging.getLogger(__name__)
+
+class RayLauncherSearchPathPlugin(SearchPathPlugin):
+    """
+    This plugin is allowing configuration files provided by the ExampleLauncher plugin to be discovered
+    and used once the ExampleLauncher plugin is installed
+    """
+
+    def manipulate_search_path(self, search_path):
+        assert isinstance(search_path, ConfigSearchPath)
+        # Appends the search path for this plugin to the end of the search path
+        search_path.append(
+            "hydra-ray-launcher", "pkg://hydra_plugins.ray_launcher.conf"
+        )
+
+def launch(*args, **kwargs):
+    setup_globals()
+    run_job(*args, **kwargs)
+
+class RayLauncher(Launcher):
+    def __init__(self):
+        self.config = None
+        self.config_loader = None
+        self.task_function = None
+
+    def setup(self, config, config_loader, task_function):
+        self.config = config
+        self.config_loader = config_loader
+        self.task_function = task_function
+        
+        if not ray.is_initialized(): ray.init(**config.select("ray.init")) if config.select("ray.init") else ray.init()
+
+    def launch(self, job_overrides):
+        """
+        :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
```

Comment:
The master branch of Hydra now requires strict python typing annotations, which is why your PR is failing lint.
Read the [contributing doc ](https://hydra.cc/docs/development/contributing)for for info (it's a bit outdated but should still be helpful).